### DISCUSSION TITLE
Debugging doesn't work on WebStorm

### DISCUSSION THREAD

**Discussion by DannyZB** [vikejs/vike#549] [Discussion]
### Description

When using the plugin none of the errors inside Vue components trigger breakpoints during SSR.
This is universal to all experiments I've run.

Is there support at all for debugging on WS?
Not sure if this is a bug report or a feature request.

**Comment by brillout** [comment]
I'm not familiar with WebStorm, so I'm not sure here. Do you think this is related to VPS or Vite? I'd guess it's related to WebStorm not supporting Vue SFCs?

**Comment by DannyZB** [comment]
Vue SFCs work well in WebStorm.
Everything works fine on the client side.
Issues are only on SSR.

Breakpoints on WS with Vite without SSR appear pointing to a proxy of the original, which doesn't enable editing in place.
That is an annoyance, not a big deal.

https://youtrack.jetbrains.com/issue/WEB-46507/Support-for-vite
This is a reference to Vite issues on JetBrains. Most of them are fixed.

**Reply by brillout** [reply]
I'm not sure how VPS can help here? I wouldn't know how it could be VPS's "fault"? I'd guess it's more about WebStorm supporting SSR than VPS related.

**Comment by DannyZB** [comment]
I'll fire up a clean SSR example and WS and let you know what happens.

**Comment by coryvirok** [comment]
I've been able to debug SSR through Webstorm but not by setting breakpoints via the UI. I was able to get the server to break using the `debugger` keyword. My setup might be different than yours, but essentially I ran `npm run dev` via webstorm's debug executor and it was able to break at the line I added the `debugger` stmt to.

Hope that helps.

**Reply by DannyZB** [reply]
It does help.
It blows my mind that they don't support something this elementary..

It's choppy  but it gave me some ability to debug. Thank you.

**Comment by brillout** [comment]
Actually, it's a Vite issue. There is work-in-progress to fix this, such as https://github.com/vitejs/vite/pull/11411.
